src/sparsezoo/v2/integration_validator.py

```diff
@@ -0,0 +1,220 @@
+            return not self.essential_model_directory_files
+
+        else:
+            # make sure that all default files in ModelDirectory are present
+            return [file for file in self.model_directory.files]
+
+    def _validate_deployment_directory(self, deployment_directory: File) -> None:
+        expected_file_names = self.integration_to_expected_training_files[
+            self.integration_name
+        ](deployment=True)
+        for file in deployment_directory.files:
+            if isinstance(file, Directory):
+                raise ValueError(
+                    "Found nested directories within `deployment` directory. "
+                    "The directory may only contain files, not directories."
+                )
+            else:
+                file_names = set(deployment_directory.get_file_names())
+                for optional_file in self.optional_training_files:
+                    file_names.discard(optional_file)
+                if expected_file_names != file_names:
+                    raise ValueError(
+                        f"Failed to find expected files "
+                        f"{expected_file_names.difference(file_names)} "
+                        f"in the `deployment` directory {deployment_directory.name}."
+                    )
+                return
+
+    def _validate_training_directory(self, training_directory: File) -> None:
+        expected_file_names = self.integration_to_expected_training_files[
+            self.integration_name
+        ]()
+        for file in training_directory.files:
+            if isinstance(file, Directory):
+                expected_name_prefix = "checkpoint_"
+                # `training directory` can either contain
+                # expected training files or
+                # sub folders with names "checkpoint_{...}",
+                # that contain expected training files
+                if not file.name.startswith(expected_name_prefix):
+                    raise ValueError(
+                        f"Found a directory in `training` directory "
+                        f"with the name: {file.name}. "
+                        f"The name of the directory should "
+                        f"start with '{expected_name_prefix}'."
+                    )
+                self._validate_training_directory(training_directory=file)
+            else:
+                file_names = set(training_directory.get_file_names())
+                for optional_file in self.optional_training_files:
+                    file_names.discard(optional_file)
+
+                if expected_file_names != file_names:
+                    raise ValueError(
+                        f"Failed to find expected files "
+                        f"{expected_file_names.difference(file_names)} "
+                        f"in the `training` directory {training_directory.name}."
+                    )
+                return
```

Comment:
See comment for `_validate_deployment_directory`

Reply:
Answered in different thread, please resolve the conversation if it addresses the issue.